.github/workflows/smi-conformance.yml

```diff
@@ -0,0 +1,19 @@
+name: SMI Conformance with Meshery
+on:
+  workflow_run:
+    workflows: ["Release Helm Chart"]
+    types:
+      - completed
```

Comment:
This workflow ideally should be run per pull request or nightly, so we don't end up discovering non-conformance during the process of cutting a release.

Reply:
We could do that, but for that, we need to publish helm charts on every nightly build. This is so because Meshery uses OSM's Helm charts to deploy it. 

So, with the current workflows, OSM's Helm charts are published during every release and this workflow runs after we make sure that the Helm charts are published and uses the latest one.

Reply:
Is it possible for Meshery to use OSM's CLI? We could publish the CLI as an artifact with the PR workflow, and the CLI could reference compatible images to install OSM for testing. 
If Helm is a hard requirement, the alternative is to figure out a solution that works as a part of the regular PR workflow.
The goal here is to avoid discovery of non-conformance **after** a release is cut. We want to be able to continuously check for conformance.

Reply:
Makes sense to run this action on every PR. We just recently moved from using OSM CLI to the Helm charts.

@DelusionalOptimist what were the caveats in using the OSM CLI? Could we do something like this?

Reply:
@shashankram this is great feedback. @shashankram, @delqn, @michelleN, we've invested so much in the SMI Conformance program to date. Before we throw more effort into the initiative (by way of enhancements), can we first incorporate use of the tool as-is, gleen some value, feedback on the tests themselves, and other aspects of Meshery's approach to SMI conformance before undertaking additional enhancements? It'll be much easier to convince contributors to poor more into this effort having reached the initial milestone of employing it (even if its only reporting after release).

What do you think?

Reply:
@leecalcote I understand your concern. My only concern is that making this a part of the release job could cause the release Github action to fail and affect the release as a whole, which is far from ideal. I think we should do the opposite - verify for conformance per PR and once OSM is conformant and the action is considered stable enough, add this is a part of the release workflow.

Reply:
You can always run Meshery and use other the UI or `mesheryctl` to run validation on an ad hoc basis.

Reply:
To clarify, my suggestion was to not run the conformance test on an ad hoc basis, but to run it more frequently (per PR, nightly) before integrating it with the release workflow.